test/binary-serializer.test.js

```diff
@@ -0,0 +1,181 @@
+    meta: require('./fixtures/escrow-finish-meta-binary.json')
+  },
+  cancel: {
+    tx: require('./fixtures/escrow-cancel-tx.json'),
+    binary: require('./fixtures/escrow-cancel-binary.json')
+  }
+}
+const PaymentChannel = {
+  create: {
+    tx: require('./fixtures/payment-channel-create-tx.json'),
+    binary: require('./fixtures/payment-channel-create-binary.json')
+  },
+  fund: {
+    tx: require('./fixtures/payment-channel-fund-tx.json'),
+    binary: require('./fixtures/payment-channel-fund-binary.json')
+  },
+  claim: {
+    tx: require('./fixtures/payment-channel-claim-tx.json'),
+    binary: require('./fixtures/payment-channel-claim-binary.json')
+  }
+}
+
+function bytesListTest () {
+  const list = new BytesList().put([0]).put([2, 3]).put([4, 5])
+  test('is an Array<Uint8Array>', function () {
+    expect(Array.isArray(list.arrays)).toBe(true)
+    expect(list.arrays[0] instanceof Uint8Array).toBe(true)
+  })
+  test('keeps track of the length itself', function () {
+    expect(list).toHaveLength(5)
+  })
+  test('can join all arrays into one via toBytes', function () {
+    const joined = list.toBytes()
+    expect(joined).toHaveLength(5)
+    expect(joined).toEqual(Uint8Array.from([0, 2, 3, 4, 5]))
+  })
+}
+bytesListTest()
+
+function assertRecycles (blob) {
+  const parser = makeParser(blob)
+  const so = parser.readType(STObject)
+  const out = new BytesList()
+  so.toBytesSink(out)
+  const hex = out.toHex()
+  expect(hex).toEqual(blob)
+  expect(hex + ':').not.toEqual(blob)
+}
+
+function nestedObjectTests () {
+  fixtures.whole_objects.forEach((f, i) => {
+    test(`whole_objects[${i}]: can parse blob and dump out same blob`,
+    /*                                              */ () => {
+        assertRecycles(f.blob_with_no_signing)
+      })
+  })
+}
+nestedObjectTests()
+
+function check (type, n, expected) {
+  test(`Uint${type.width * 8} serializes ${n} as ${expected}`, function () {
+    const bl = new BytesList()
+    const serializer = new BinarySerializer(bl)
+    if (expected === 'throws') {
+      expect(() => serializer.writeType(type, n)).toThrow()
+      return
+    }
+    serializer.writeType(type, n)
+    expect(bl.toBytes()).toEqual(Uint8Array.from(expected))
+  })
+}
+
+check(UInt8, 5, [5])
+check(UInt16, 5, [0, 5])
+check(UInt32, 5, [0, 0, 0, 5])
+check(UInt32, 0xFFFFFFFF, [255, 255, 255, 255])
+check(UInt8, 0xFEFFFFFF, 'throws')
+check(UInt16, 0xFEFFFFFF, 'throws')
+check(UInt16, 0xFEFFFFFF, 'throws')
+check(UInt64, 0xFEFFFFFF, [0, 0, 0, 0, 254, 255, 255, 255])
+check(UInt64, -1, 'throws')
+check(UInt64, 0, [0, 0, 0, 0, 0, 0, 0, 0])
+check(UInt64, 1, [0, 0, 0, 0, 0, 0, 0, 1])
+check(UInt64, new BN(1), [0, 0, 0, 0, 0, 0, 0, 1])
+
+// function parseLedger4320278() {
+//   test('can parse object', done => {
+//     const json = loadFixture('as-ledger-4320278.json');
+//     json.forEach(e => {
+//       assertRecycles(e.binary);
+//     });
+//     done();
+//   });
+// }
+// parseLedger4320278();
```

Comment:
This is super weird. The fixture `'as-ledger-4320278.json'` does not exist in `./test/fixtures`. Not quite sure exactly whats going on here.

Reply:
I commented out `function parseLedger4320278()` because it could not load the nonexistent fixture

Reply:
Thanks for pointing this out. The prior test file skipped the test that used this function: `describe.skip`. The line originates from the initial commit of this repo (5 years ago). No worries